Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -12446,6 +12461,15 @@ void WebPageProxy::shouldAllowDeviceOrientationAndMotionAccess(IPC::Connection&
 
     protectedWebsiteDataStore()->protectedDeviceOrientationAndMotionAccessController()->shouldAllowAccess(*this, *frame, WTFMove(frameInfo), mayPrompt, WTFMove(completionHandler));
 }
+
+bool WebPageProxy::originHasDeviceOrientationAndMotionAccess(const WebCore::SecurityOriginData& origin)
+{
+    if (!protectedPreferences()->deviceOrientationPermissionAPIEnabled())
+        return true;
```

Comment:
Should this return `true` in this case?

Reply:
Yes, if the API is not enabled, that means site doesn't need permission to proceed, so we skip the origin check by making all origins are allowed.